geode-core/src/main/java/org/apache/geode/internal/tcp/Connection.java

```diff
@@ -2846,6 +2846,9 @@ private void processInputBuffer(AbstractExecutor threadMonitorExecutor)
                       "Allocating larger network read buffer, new size is {} old size was {}.",
                       allocSize, oldBufferSize);
                   inputBuffer = inputSharing.expandReadBufferIfNeeded(allocSize);
+                  // we're returning to the caller (done == true) so make buffer writable
```

Comment:
If we have to comment it to make it readable perhaps it deserves a well named method.
```java
...
makeWriteable(inputBuffer);
...

void makeWriteable(final ByteByffer buffer) {
  buffer.position(buffer.limit());
  buffer.limit(buffer.capacity());
}
```
Now this method is unit testable too.